### PR TITLE
 removed redundant code from __main__.py used in refactored for Agent class

### PR DESCRIPTION
**Pull request by adtrem**
### Changes

- To improve code readability and maintenance, I refactored the Agent class by:
    - Separating the code into several small methods.
    - Giving each method a clear name and specific responsibilities.
- This makes the code easier for future developers to understand and maintain, as they can easily identify specific functions of the code and see how they are related to the overall program.

- Removed redundant code from __main__.py file that is already used in agent.py file.

- Use an instance of the Agent class from the agent.py file in the __main__.py file.

### PR Quality Checklist
- [x] My pull request is atomic and focuses on a single change.
- [x] I have thoroughly tested my changes with multiple different prompts.
- [x] I have considered potential risks and mitigations for my changes.
- [x] I have documented my changes clearly and comprehensively.
- [x] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

File: autogpt/agent.py [hunk 1]
```diff
@@ -26,163 +26,178 @@ class Agent:
...
-            else:
-                self.full_message_history.append(
-                    create_chat_message("system", "Unable to execute command")
-                )
-                logger.typewriter_log(
-                    "SYSTEM: ", Fore.YELLOW, "Unable to execute command"
-                )
+            self.check_limit_loop_count()
+
+    def start_argent(self):
```
Comment (harrymunro) [inline autogpt/agent.py]: Should this be start_agent?

```suggestion
    def start_agent(self):
```

File: autogpt/agent.py [hunk 2]
```diff
@@ -26,163 +26,178 @@ class Agent:
...
-                if self.next_action_count > 0:
-                    self.next_action_count -= 1
+            self.loop_count += 1
 
-            memory_to_add = (
-                f"Assistant Reply: {assistant_reply} "
-                f"\nResult: {result} "
-                f"\nHuman Feedback: {self.user_input} "
-            )
+            response = self.start_argent()
```
Comment (harrymunro) [inline autogpt/agent.py]: Typo?

```suggestion
            response = self.start_agent()
```

Comment (adtrem) [reply autogpt/agent.py]: yes it's a typo

**Comment by nponeccop** [comment]
@adtrem There are conflicts now

**Comment by adtrem** [comment]
The merge conflicts have been resolved.

**Review by nponeccop — APPROVED** [review]

**Comment by p-i-** [comment]
This is a mass message from the AutoGPT core team.
Our apologies for the ongoing delay in processing PRs.
This is because we are re-architecting the AutoGPT core!

For more details (and for infor on joining our Discord), please refer to:
  https://github.com/Significant-Gravitas/Auto-GPT/wiki/Architecting

**Comment by Pwuts** [comment]
I'm sorry to see this PR didn't go all the way when it could. By now there are lots of conflicts, and Chopping up the agent's main loop into smaller pieces is now a core part of #4799. Still, thanks a lot for submitting! I will be looking to this PR for some inspiration to finish #4799. :)

If you are still interested in contributing, please join the [discord] and ping us there. That way we can coordinate and make sure we don't duplicate anyone's efforts.

[discord]: https://discord.gg/autogpt

**Pull request closed without merging**